mesonbuild/mintro.py

```diff
@@ -178,9 +178,10 @@ def list_targets(builddata: build.Build, installdata, backend: backends.Backend)
     return tlist
 
 def list_buildoptions_from_source(intr: IntrospectionInterpreter) -> List[Dict[str, Union[str, bool, int, List[str]]]]:
-    return list_buildoptions(intr.coredata)
+    subprojects = [i['name'] for i in intr.project_data['subprojects']]
+    return list_buildoptions(intr.coredata, subprojects)
 
-def list_buildoptions(coredata: cdata.CoreData) -> List[Dict[str, Union[str, bool, int, List[str]]]]:
+def list_buildoptions(coredata: cdata.CoreData, subprojects: List[str] = None) -> List[Dict[str, Union[str, bool, int, List[str]]]]:
```

Comment:
The type is wrong, you can't assign a None to a List[str], I think this should be Optional[List[str]]

Reply:
Done.